src/internal/queues.mjs

```diff
@@ -1,66 +1,104 @@
 class QueueItem {
   constructor(data) {
     this.data = data;
-    this.previous = null;
+    this.next = null;
```

Comment:
This has been bothering me for ages. If you are at the head of a queue of people, the person behind you is the next person in line, not the previous person in line. The previous person in line is no longer in the line.